### ISSUE TITLE
(Almost) all the comments and file headers are useless

### ISSUE THREAD

**Issue by lemmit** [cqfn/eo#140]
The signal vs noise ratio in the source files is pretty high. 
Comments contain mostly redundant information, derived from types/variable names. My suggestion is to remove them too, unless they provide some real additional information.
Also have you considered removing the license headers? In my experience that is pretty much an anti-pattern (and thing against clean code rules).

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this issue

**Comment by yegor256** [comment]
@lemmit thanks for the ticket, but look at this: https://softwareengineering.stackexchange.com/questions/125836/do-you-have-to-include-a-license-notice-with-every-source-file It seems that we do need those license boilerplate texts in each file. For many reasons. As for the JavaDoc at variables and types: we're following the rules of qulice: www.qulice.com

**Comment by lemmit** [comment]
It seems that the license can be shortened into single-line notice with note where to find the full license.

Is there ANY other reason to keep those comments? "We do it because static-checker says so"?
Good naming should eliminate the need of having comments. Shouldn't it be a main goal?

**Comment by pa9ey** [comment]
@lemmit We've debated this before (and other similar things) in #119 ( #107 #108 #109 ) and I'm very sympathetic to your position. But I think I ended up agreeing that the license was necessary in every file but that the full text shouldn't be included. I guess if it is included, I would still want it at the end of the file. It's such a horrible messy distraction to see it at the start of every single file.
